pkg/comp-functions/functions/common/mailgun_alerting.go

```diff
@@ -0,0 +1,161 @@
+package common
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"strconv"
+
+	xkube "github.com/crossplane-contrib/provider-kubernetes/apis/object/v1alpha1"
+	alertmanagerv1alpha1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1alpha1"
+	runtime "github.com/vshn/appcat/v4/pkg/comp-functions/runtime"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/pointer"
+	controllerruntime "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func MailgunAlerting(obj client.Object) func(ctx context.Context, iof *runtime.Runtime) runtime.Result {
+	return func(ctx context.Context, iof *runtime.Runtime) runtime.Result {
+		log := controllerruntime.LoggerFrom(ctx)
+		log.Info("Starting mailgun-alerting function")
+
+		err := iof.Observed.GetComposite(ctx, obj)
+		if err != nil {
+			return runtime.NewFatalErr(ctx, "an oopsie happened", err)
+		}
+		alertConfig, ok := obj.(Alerter)
+		if !ok {
+			return runtime.NewWarning(ctx, fmt.Sprintf("Type %s doesn't implement Alerter interface", reflect.TypeOf(obj).String()))
+		}
+
+		email := alertConfig.GetVSHNMonitoring().Email
+		instanceNamespace := alertConfig.GetInstanceNamespace()
+		name := obj.GetName()
+
+		if instanceNamespace == "" {
+			return runtime.NewWarning(ctx, "Instance namespace not yet populated, skipping")
+		}
+
+		if !mailAlertingEnabled(iof.Config) {
+			if email == "" {
+				return runtime.NewNormal()
+			}
+			return runtime.NewWarning(ctx, "Email Alerting is not enabled")
+		}
+		// Wait for the next reconciliation in case instance namespace is missing
+		if instanceNamespace == "" {
+			return runtime.NewWarning(ctx, "Composite is missing instance namespace, skipping transformation")
+		}
+
+		if email == "" {
+			return runtime.NewNormal()
+		}
```

Comment:
```suggestion
		if email == "" {
			return runtime.NewNormal()
		}
		if !mailAlertingEnabled(iof.Config) {
			return runtime.NewWarning(ctx, "Email Alerting is not enabled")
		}
		// Wait for the next reconciliation in case instance namespace is missing
		if instanceNamespace == "" {
			return runtime.NewWarning(ctx, "Composite is missing instance namespace, skipping transformation")
		}

```

Reply:
Also don't forget to clean up your commit history